src/test/java/uk/gov/hmcts/reform/professionalapi/service/LegacyPbaAccountServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+
+        professionalUserServiceImpl = new ProfessionalUserServiceImpl(
+                organisationRepository,
+                professionalUserRepository,
+                userAttributeRepository,
+                prdEnumRepository,
+                userAttributeService);
+
+
+        List<PaymentAccount>  paymentAccountsFromEntity = new ArrayList<>();
+
+        when(professionalUserServiceImpl.findProfessionalUserByEmailAddress(any(String.class)))
+                .thenReturn(professionalUser);
+
+        when(professionalUser.getOrganisation()).thenReturn(organisation);
+
+        when(organisation.getPaymentAccounts()).thenReturn(paymentAccounts);
+
+        when(professionalUser.getUserAccountMap()).thenReturn(userAccountMaps);
+
+        //when(legacyPbaAccountServiceImpl.getPaymentAccountsFromUserAccountMap(professionalUser.setUserAccountMap())).thenReturn(paymentAccountsFromEntity);
+
+        when(legacyPbaAccountServiceImpl.findLegacyPbaAccountByUserEmail(any(ProfessionalUser.class)))
+                .thenReturn(paymentAccountPbaNumbers);
+
+    }
+
+    @Test
+    public void findLegacyPbaAccountByUserEmailTest() {
+
+        ProfessionalUser professionalUser = professionalUserServiceImpl.findProfessionalUserByEmailAddress("some-email");
+
+        assertThat(professionalUser).isNotNull();
+
+        paymentAccountPbaNumbers = legacyPbaAccountServiceImpl.findLegacyPbaAccountByUserEmail(professionalUser);
+
+        assertThat(paymentAccountPbaNumbers).isNotNull();
+        verify(
+                professionalUserRepository,
+                times(1)).findByEmailAddress(any(String.class));
+
+        verify(
+                legacyPbaAccountServiceImpl,
+                times(1)).findLegacyPbaAccountByUserEmail(any(ProfessionalUser.class));
+
+    }
+
+    @Test
+    public void findLegacyPbaAccountsByUserEmailTest() {
+
+        Organisation theOrganisation = new Organisation("some-org-", OrganisationStatus.PENDING, "sra-id", "company-number", false, "company-url");
+
+        ProfessionalUser theSuperUser = new ProfessionalUser("some-fname", "some-lname", "some-email", ProfessionalUserStatus.PENDING, theOrganisation);
+        theOrganisation.addProfessionalUser(theSuperUser);
+
+        ContactInformation theContactInfo = new ContactInformation("addressLine-1", "addressLine-2", "addressLine-3", "townCity", "county", "country", "postCode", theOrganisation);
+        theOrganisation.addContactInformation(theContactInfo);
+
+        PaymentAccount thePaymentAcc = new PaymentAccount("pbaNumber-1");
+        theOrganisation.addPaymentAccount(thePaymentAcc);
+
+        organisationRepository.save(theOrganisation);
+
+        ProfessionalUser professionalUser = professionalUserServiceImpl.findProfessionalUserByEmailAddress("some-email");
+
+        List<String> paymentAccountPbaNumbers = legacyPbaAccountServiceImpl.findLegacyPbaAccountByUserEmail(professionalUser);
+
+        assertThat(paymentAccountPbaNumbers).isNotNull();
+
```

Comment:
Could be improved to include more detailed assertions. For instance that the correct status code is being returned etc.